L/LLVM/common.jl

```diff
@@ -273,6 +273,14 @@ ninja -j${nproc} -vv
 # Install!
 ninja install
 
+shopt -s nullglob
+for x in ${prefix}/lib/*.so; do
+    # All things that have RPATH or doesn't have $ORIGIN,
+    # but not those without $RUNPATH and $ORIGIN
+    [ -n "$(objdump -x "$x" | grep 'RPATH\|RUNPATH')" ] && \
+      patchelf --set-rpath "\$ORIGIN" "$x";
+done
```

Comment:
Also here, the fact that [we have](https://github.com/JuliaPackaging/Yggdrasil/issues/3703#issuecomment-1290329832) `$ORIGIN/../lib` as RPATH makes me think this should have always been `$ORIGIN` (library is in `${prefix}/lib`, so `$ORIGIN/../lib` is just `$ORIGIN`).  Maybe when we [normalise the rpaths](https://github.com/JuliaPackaging/BinaryBuilder.jl/blob/23847acbc1ecf0c1a7cdc09ad657036776d749f7/src/auditor/dynamic_linkage.jl#L454-L485) we should flatten relative paths, so that these identities becomes clear and we can simplify the paths we put there.

Reply:
the library may later get moved to `lib/julia`, resulting in breakage for people linking against it

that was fixed in Julia v1.7 (https://github.com/JuliaLang/julia/pull/42788), but strictly only on that release branch (@vchuravy)